triggers/daml/AutoApproval.daml

```diff
@@ -167,7 +167,11 @@ generateCommands choiceArg queryResults = map (\(cid, _) -> (toAnyContractId cid
 generateCommandsWithCid : forall t c r t2. (Template t, Choice t c r, Template t2) => (t2 -> t -> Bool) -> (ContractId t2 -> c) -> [(ContractId t2, t2)] -> [(ContractId t, t)] -> [(AnyContractId, Command)]
 generateCommandsWithCid _ _ [] _ = []
 generateCommandsWithCid _ _ _ [] = []
-generateCommandsWithCid pred toChoice requests targets = mapOptional (\(reqCid, req) -> (\(cid, _) -> (toAnyContractId cid, exerciseCmd cid $ toChoice reqCid)) <$> findFirstMatch req targets pred) requests
+generateCommandsWithCid pred toChoice requests targets = mapOptional (\(reqCid, req) -> getCommandWithCid toChoice reqCid <$> findFirstMatch req targets pred) requests
+
+-- | Takes a choice and a contract id which is a parameter of that choice, the contract which exposes this choice and builds up a command
+getCommandWithCid : forall t c r t2. (Template t, Choice t c r, Template t2) => (ContractId t2 -> c) -> ContractId t2 -> (ContractId t, t) -> (AnyContractId, Command)
+getCommandWithCid toChoice requestCid (cid, _) = (toAnyContractId cid, exerciseCmd cid $ toChoice requestCid)
```

Comment:
The goal of doing this is to make it easier to comprehend what this function is doing, by removing one of the lambda's out into its own function.

Reply:
What I mean is, can you just remove `forall t c r t2. (Template t, Choice t c r, Template t2) =>`, or parts of it? Do you really need that type constraint?

Reply:
The best I can reduce it the type constraint to without an error is `forall t c r. Choice t c r =>`.

Reply:
I've updated this signature and a few others that could also be simplified.